big_tests/tests/rest_client_SUITE.erl

```diff
@@ -424,6 +433,81 @@ user_can_be_added_and_removed_by_room_jid(Config) ->
         ?assertEqual(<<"204">>, Status)
     end).
 
+msg_with_props_is_sent_and_delivered_over_xmpp(Config) ->
+    escalus:fresh_story(Config, [{alice, 1}, {bob, 1}], fun(Alice, Bob) ->
+        BobJID = user_jid(Bob),
+        MsgID = base16:encode(crypto:strong_rand_bytes(5)),
+        M1 = rest_helper:make_msg_stanza_with_props(BobJID,MsgID),
+
+        escalus:send(Alice, M1),
+
+        M2 = escalus:wait_for_stanza(Bob),
+        escalus:assert(is_message, M2)
+    end).
+
+msg_with_props_can_be_parsed(Config) ->
+    escalus:fresh_story(Config, [{alice, 1}, {bob, 1}], fun(Alice, Bob) ->
+        AliceJID = escalus_utils:jid_to_lower(escalus_client:short_jid(Alice)),
+        BobJID = escalus_utils:jid_to_lower(escalus_client:short_jid(Bob)),
+        MsgID = base16:encode(crypto:strong_rand_bytes(5)),
+        M1 = rest_helper:make_msg_stanza_with_props(BobJID,MsgID),
+
+        escalus:send(Alice, M1),
+        mam_helper:maybe_wait_for_archive(Config),
+        
+        AliceCreds = {AliceJID, user_password(alice)},
+
+        % recent msgs with a limit
+        M2 = get_messages_with_props(AliceCreds, BobJID, 1),
+
+        [{MsgWithProps}|_] = M2,
+
+        Data = maps:from_list(MsgWithProps),
+
+        #{<<"properties">>:={Props}} = Data,
+        #{<<"id">>:={ReceivedMsgID}} = Data,
+
+        %we are expecting two properties:"some_string" and "some_number" for this test message
+        %test message defined in rest_helper:make_msg_stanza_with_props
+        2 = length(Props),
+        ReceivedMsgID = MsgID
+
+    end).
+
+msg_with_malformed_props_is_sent_and_delivered_over_xmpp(Config) ->
+    escalus:fresh_story(Config, [{alice, 1}, {bob, 1}], fun(Alice, Bob) ->
+        BobJID = user_jid(Bob),
+        MsgID = base16:encode(crypto:strong_rand_bytes(5)),
+
+        M1 = rest_helper:make_malformed_msg_stanza_with_props(BobJID,MsgID),
+
+        escalus:send(Alice, M1),
+
+        M2 = escalus:wait_for_stanza(Bob),
+        escalus:assert(is_message, M2)
+    end).
+
+msg_with_malformed_props_can_be_parsed(Config) ->
+    escalus:fresh_story(Config, [{alice, 1}, {bob, 1}], fun(Alice, Bob) ->
+        AliceJID = escalus_utils:jid_to_lower(escalus_client:short_jid(Alice)),
+        AliceCreds = {AliceJID, user_password(alice)},
+        BobJID = escalus_utils:jid_to_lower(escalus_client:short_jid(Bob)),
+        MsgID = base16:encode(crypto:strong_rand_bytes(5)),
+
+        M1 = rest_helper:make_malformed_msg_stanza_with_props(BobJID,MsgID),
+        escalus:send(Alice, M1),
```

Comment:
@michalwski thanks for review.

It seems to me that sending raw stanza direct to `Alice` takes long time than expected.  Either `mam_helper:maybe_wait_for_archive(Config)` not work or waiting time too low.

Reply:
First of all you can use `escalus:wait_for_stanza(Bob)` to be sure Bob gets the message. Then instead of `mam_helper"maybe_wait_for_archive/1` you can use `mam_helper:wait_for_archive_size`. Which waits until the archive is of expected size. There is 20s timeout for this in case the archive never reaches the expected size.

Also it's important to know that not all jobs on travis are configured with mod_mam enabled (for instance `internal_mnesia`). That's why test cases working with MAM in this suite needs to be skipped. You can do it by extending the list `MAMTestCases` in `init_per_testcase`.